packages/tools/client-debugger/client-debugger-view/src/components/DataObjectsView.tsx

```diff
@@ -18,10 +32,61 @@ export type DataObjectsViewProps = HasContainerId;
  * Dispatches data object rendering based on those provided view {@link DataObjectsViewProps.renderOptions}.
  */
 export function DataObjectsView(props: DataObjectsViewProps): React.ReactElement {
+	const { containerId } = props;
+
+	const messageRelay = useMessageRelay();
+
+	const [rootDataHandles, setRootDataHandles] = React.useState<
+		Record<string, RootHandleNode> | undefined
+	>();
+
+	React.useEffect(() => {
+		const inboundMessageHandlers: InboundHandlers = {
+			["ROOT_DATA_VISUALIZATIONS"]: (untypedMessage) => {
+				const message: RootDataVisualizationsMessage =
+					untypedMessage as RootDataVisualizationsMessage;
+
+				if (message.data.containerId === containerId) {
+					setRootDataHandles(message.data.visualizations);
+
+					return true;
+				} else {
+					return false;
+				}
+			},
+		};
+
+		function messageHandler(message: Partial<IDebuggerMessage>): void {
+			handleIncomingMessage(message, inboundMessageHandlers, {
+				context: loggingContext,
+			});
+		}
+
+		messageRelay.on("message", messageHandler);
+
+		messageRelay.postMessage({
+			type: "GET_ROOT_DATA_VISUALIZATIONS",
+			data: {
+				containerId,
+			},
+		});
+
+		return (): void => {
+			messageRelay.off("message", messageHandler);
+		};
+	}, [containerId, rootDataHandles, messageRelay]);
+
+	if (rootDataHandles === undefined) {
+		return <Waiting label="Waiting for container DDS data. DATAOBJECTSVIEW" />;
+	}
+
 	return (
-		<div className="data-objects-view">
-			<h3>Container Data</h3>
-			<div>TODO: data visualization is not yet supported.</div>
+		<div>
+			<>
```

Comment:
Do we need this fragment between the div and the content? I realize you're still working on the view aspect here, so feel free to disregard, but generally fragments are only needed to wrap multiple children when you don't want to introduce an actual wrapper component. But you already have a div here, so adding the fragment shouldn't do anything.

Reply:
No fragment is unnecessary. I did not remove it in my first commit as the work was in progress.

It's now removed!